application/client/src/app/ui/views/toolbar/chart/chart/state.ts

```diff
@@ -39,8 +45,54 @@ export class State extends AdvancedState {
     }
 
     public onClick(event: MouseEvent) {
+        const position: number = this._calculatePosition(event);
+        position >= 0 && this._session.cursor.select(position, Owner.Chart);
+    }
+
+    public onMouseMove(event: MouseEvent) {
+        const rect: DOMRect = this._element.getBoundingClientRect();
+        const offsetX = event.offsetX;
+        if (offsetX > 0) {
+            this.crosshairLeft = offsetX;
+            this.showLeftTooltip = offsetX <= rect.width / 2;
+        }
+        if (event.offsetY > 0) {
+            this.tooltipTop = rect.height / 2;
+            const position: number = this._calculatePosition(event);
+            this.tooltip = position >= 0 ? `${position}` : '';
+        }
+    }
+
+    public onMouseEnter(mouseEnter: boolean) {
+        if (!mouseEnter) {
+            this.tooltip = '';
+        }
+        this.mouseEnter = mouseEnter;
+    }
+
+    protected _resize() {
+        this._parent.detectChanges();
+    }
+
+    protected _fetch(width: number, range: IRange): Promise<void> {
+        this._loading = true;
+        return this._session.search
+            .getScaledMap(width, range)
+            .then((map) => {
+                this._map = map;
+                this._draw(EChartName.chartFilters);
+            })
+            .catch((err: Error) => {
+                this._parent.log().error(err);
+            })
+            .finally(() => {
+                this._loading = false;
+            });
+    }
+
+    private _calculatePosition(event: MouseEvent): number {
```

Comment:
In general, this method isn't trustable. Could it be NaN? or infinitive? If it's possible, I would use as output `number | Error` to make it more predictable

Reply:
I'm not sure at which place we could get something else than a number. I mean I make sure that we always calculate the position without failure, or did I miss something?

Reply:
I see it now, yeah I've changed the code accordingly.